src/components/flyout/deviceDeletionFlyout.js

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import * as actions from '../../actions';
+import Spinner from '../spinner/spinner';
+import Delete from '../../assets/icons/Delete_dark.svg';
+import CancelX from '../../assets/icons/CancelX.svg';
+import TooltipSvg from '../../assets/icons/Tooltip.svg';
+import ApiService from '../../common/apiService';
+import lang from '../../common/lang';
+import PcsBtn from '../shared/pcsBtn/pcsBtn';
+import SummarySection from '../shared/summarySection/summarySection';
+
+import './deviceDeletionFlyout.css';
+
+class DeviceDeletionFlyout extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      showSpinner: false,
+      deviceDeleted: false,
+      IsSimulated: true
+    };
+
+    this.onConfirm = this.onConfirm.bind(this);
+  }
+
+  componentDidMount() {
+    const { devices } = this.props;
+    this.setState({
+      IsSimulated: devices.some(({ IsSimulated }) => IsSimulated)
+    })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { devices } = nextProps;
+    this.setState({
+      IsSimulated: devices.some(({ IsSimulated }) => IsSimulated)
+    })
+    if (devices.length !== this.props.devices.length) {
+      this.setState({
+        showSpinner: false,
+        deviceDeleted: false
+      });
+    }
+  }
+
+  onConfirm() {
+    const { devices } = this.props;
+    const physicalDevices = devices.filter(({ IsSimulated }) => !IsSimulated);
```

Comment:
Either save the filtered devices in the state or pass the values from the render into the onConfirm.